homeassistant/components/device_tracker/ubus.py

```diff
@@ -11,25 +11,27 @@
 from datetime import timedelta
 
 import requests
+import voluptuous as vol
 
-from homeassistant.components.device_tracker import DOMAIN
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import DOMAIN, PLATFORM_SCHEMA
 from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
-from homeassistant.helpers import validate_config
 from homeassistant.util import Throttle
 
 # Return cached results if last scan was less then this time ago.
 MIN_TIME_BETWEEN_SCANS = timedelta(seconds=5)
 
 _LOGGER = logging.getLogger(__name__)
 
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string
```

Comment:
I'll do it when I add the Automatic as a sensor (future feature).

On Fri, Sep 2, 2016 at 4:17 PM, Johann Kellerman <notifications@github.com>
wrote:

> In homeassistant/components/device_tracker/ubus.py
> <https://github.com/home-assistant/home-assistant/pull/3125#discussion_r77419303>
> :
>
> >  from homeassistant.util import Throttle
> >
> >  # Return cached results if last scan was less then this time ago.
> >  MIN_TIME_BETWEEN_SCANS = timedelta(seconds=5)
> >
> >  _LOGGER = logging.getLogger(__name__)
> >
> > +PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
> > +    vol.Required(CONF_HOST): cv.string,
> > +    vol.Required(CONF_PASSWORD): cv.string,
> > +    vol.Required(CONF_USERNAME): cv.string
>
> Could use cv.string_schema helper from #3123
> <https://github.com/home-assistant/home-assistant/pull/3123>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/3125/files/7f21ec2403c4215dfc4591bb5ded4fa2b5c07ad3#r77419303>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC2fZac4UsCdQ2SJCJuUZaEVbK8ZSa5Hks5qmKCQgaJpZM4JzLSx>
> .
>